packages/modules/common/component_state.py

```diff
@@ -71,7 +71,11 @@ def __init__(
         if _check_none(currents):
             currents = [0.0]*3
         else:
-            if not ((sum(currents) < 0 and power < 0) or (sum(currents) > 0 and power > 0)):
+            if not (
+                (sum(currents) < 0 and power < 0) or
+                (sum(currents) > 0 and power > 0) or
+                (sum(currents) == 0 and power == 0)
+            ):
                 log.debug("currents sign wrong "+str(currents))
```

Comment:
wie wäre es so?


```python
if not any([
    sum(currents) < 0 and power < 0,
    sum(currents) > 0 and power > 0,
    sum(currents) == 0 and power == 0
]):
```

Und da es schon 2 mal auf taucht würde ich es in eine eigene Funktion auslagern.

Reply:
@seaspotter  Kannst Du bitte wie von @tpd-opitz  vorgeschlagen, die Prüfung in eine Funktion auslagern?